File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/B_Robot_Arms.cpp
#include <bits/stdc++.h>
#define MAX_N 100'000
#define INF 1'000'000'000'000'000'000
using namespace std;
typedef long long ll;
typedef pair<ll, ll> pll;

ll N;
ll X[MAX_N], L[MAX_N];

void solve() {
  vector<pll> V;
  for (int i = 0; i < N; ++i) {
    V.push_back({X[i] + L[i], X[i] - L[i]});
  }
  sort(V.begin(), V.end());
  ll res = 0;
  ll t = -INF;
  for (int i = 0; i < N; ++i) {
    pll v = V[i];
    if (t <= v.second) {
      t = v.first;
      ++res;
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N;
  for (int i = 0; i < N; ++i) {
    cin >> X[i] >> L[i];
  }
  solve();
}

File: AtCoder/ACP/BootCamp4b/Hard100_2/python/B_Robot_Arms.py
INF = float('inf')
N = int(input())
V = []
for _ in range(N):
    X, L = map(int, input().split())
    V.append((X + L, X - L))
V.sort()
res = 0
t = -INF
for v in V:
    if t <= v[1]:
        t = v[0]
        res += 1
print(res)
